ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -1192,6 +1192,10 @@ private static void createReplImportTasks(
       // have been already created when replaying previous events. So no need to create table
       // again.
       if (x.getEventType() != DumpType.EVENT_COMMIT_TXN) {
+        //Don't set location for managed tables while creating the table.
+        if (x.getEventType() == DumpType.EVENT_CREATE_TABLE && !tblDesc.isExternal()) {
```

Comment:
We needn't handle the external table?

Reply:
For external table, we can still set location. Managed tables are hive managed. So this is not allowed.